pkg/api/utils/secretUtils.go

```diff
@@ -85,24 +88,70 @@ func (s *SecretHandler) getHTTPClient() *http.Client {
 }
 
 // CreateSecret creates a new secret
-func (s *SecretHandler) CreateSecret(secret models.Secret) (string, *models.Error) {
+func (s *SecretHandler) CreateSecret(secret models.Secret) error {
 	body, err := json.Marshal(secret)
 	if err != nil {
-		return "", buildErrorResponse(err.Error())
+		return err
 	}
-	return post(s.Scheme+"://"+s.BaseURL+v1SecretPath, body, s)
+	_, errObj := post(s.Scheme+"://"+s.BaseURL+v1SecretPath, body, s)
+	if errObj != nil {
+		return errors.New(*errObj.Message)
+	}
+	return nil
 }
 
 // UpdateSecret creates a new secret
-func (s *SecretHandler) UpdateSecret(secret models.Secret) (string, *models.Error) {
+func (s *SecretHandler) UpdateSecret(secret models.Secret) error {
 	body, err := json.Marshal(secret)
 	if err != nil {
-		return "", buildErrorResponse(err.Error())
+		return err
+	}
+	_, errObj := put(s.Scheme+"://"+s.BaseURL+v1SecretPath, body, s)
+	if errObj != nil {
+		return errors.New(*errObj.Message)
 	}
-	return put(s.Scheme+"://"+s.BaseURL+v1SecretPath, body, s)
+	return nil
 }
 
 // DeleteSecret deletes a secret
-func (s *SecretHandler) DeleteSecret(secretName, secretScope string) (string, *models.Error) {
-	return delete(s.Scheme+"://"+s.BaseURL+v1SecretPath+"?name="+secretName+"&scope="+secretScope, s)
+func (s *SecretHandler) DeleteSecret(secretName, secretScope string) error {
+	_, err := delete(s.Scheme+"://"+s.BaseURL+v1SecretPath+"?name="+secretName+"&scope="+secretScope, s)
+	if err != nil {
+		return errors.New(*err.Message)
+	}
+	return nil
+}
+
+// GetSecrets returns a list of created secrets
+func (s *SecretHandler) GetSecrets() (*models.GetSecretsResponse, error) {
+	req, err := http.NewRequest("GET", s.Scheme+"://"+s.BaseURL+v1SecretPath, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	addAuthHeader(req, s)
+
+	resp, err := s.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		errObj := &models.Error{}
+		if err := json.Unmarshal(body, errObj); err != nil {
```

Comment:
we should re-add the methods for (de)serializing to the structs itself. I've already done this for the e.g. here: https://github.com/keptn/go-utils/blob/21f45c833164c75788f2bd93e02ae9acd54452cb/pkg/api/models/uniform.go#L67

but lets do this in a separate PR for all models?

Reply:
agreed, let's add that afterwards